x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/task/rule_migrations_task_evaluator.ts

```diff
@@ -65,4 +73,87 @@ export class RuleMigrationTaskEvaluator extends RuleMigrationTaskRunner {
         this.logger.error(`Evaluation error:\n ${JSON.stringify(err, null, 2)}`);
       });
   }
+
+  private getEvaluators(): Evaluator[] {
+    return Object.entries(this.evaluators).map(([key, evaluator]) => {
+      return (args) => {
+        const result = evaluator(args);
+        return { key, ...result };
+      };
+    });
+  }
+
+  /**
+   * This is a map of custom evaluators that are used to evaluate rule migration tasks
+   * The object keys are used for the `key` property of the evaluation result, and the value is a function that takes a the `run` and `example`
+   * and returns a `score` and a `comment` (and any other data needed for the evaluation)
+   **/
+  private readonly evaluators: Record<string, CustomEvaluator> = {
+    translation_result: ({ run, example }) => {
+      const runResult = (run?.outputs as MigrateRuleState)?.translation_result;
+      const expectedResult = (example?.outputs as MigrateRuleState)?.translation_result;
+
+      if (!expectedResult) {
+        return { comment: 'No translation result expected' };
+      }
+      if (!runResult) {
+        return { score: false, comment: 'No translation result received' };
+      }
+
+      if (runResult === expectedResult) {
+        return { score: true, comment: 'Correct' };
+      }
+
+      return {
+        score: false,
+        comment: `Incorrect, expected "${expectedResult}" but got "${runResult}"`,
+      };
+    },
+
+    custom_query_accuracy: ({ run, example }) => {
+      const runQuery = (run?.outputs as MigrateRuleState)?.elastic_rule?.query;
+      const expectedQuery = (example?.outputs as MigrateRuleState)?.elastic_rule?.query;
+
+      if (!expectedQuery) {
+        return { comment: 'No custom translation expected' };
```

Comment:
As I mentioned in case of `runPrebuiltRuleId`, I think if `expectedQuery` is undefined, then we should assert that `runQuery` is also the same.

Reply:
Makes sense, added this scenario here https://github.com/elastic/kibana/pull/219769/commits/6a7381b6a126b45f029b66fe9d0555e7fe6b06ce